apps/web/src/components/tx/ConfirmTxDetails/NameChip.tsx

```diff
@@ -22,15 +23,16 @@ const NameChip = ({
   const name = toName || contractInfo?.name
   const logo = toLogo || contractInfo?.logoUri
 
+  const addressBook = useAddressBook()
+  const isInAddressBook = toAddress !== undefined && !!addressBook[toAddress]
+  const isUntrusted = !isInAddressBook && contractInfo.isUnverifiedContract
```

Comment:
An optimization would be to add the address book name to the `toName` definition a few lines above. Then it will skip the requests for a name/contract info in the `useAddressName` hook.